client/driver/docker.go

```diff
@@ -893,4 +852,26 @@ func (h *DockerHandle) run() {
 		h.logger.Printf("[ERR] driver.docker: failed to kill the syslog collector: %v", err)
 	}
 	h.pluginClient.Kill()
+
+	// Stop the container just incase the docker daemon's wait returned
+	// incorrectly
+	if err := h.client.StopContainer(h.containerID, 0); err != nil {
+		_, noSuchContainer := err.(*docker.NoSuchContainer)
+		_, containerNotRunning := err.(*docker.ContainerNotRunning)
+		if !containerNotRunning && !noSuchContainer {
+			h.logger.Printf("[ERR] driver.docker: error stopping container: %v", err)
+		}
+	}
+
+	// Remove the container
+	if err := h.client.RemoveContainer(docker.RemoveContainerOptions{ID: h.containerID, Force: true}); err != nil {
```

Comment:
Wrap this in `if h.cleanupContainer {`

The user can configure Nomad to keep the container

Reply:
It looks like you didn't do this... Is there any way to get Nomad to keep a container that has exited? There is a reason to keep a container once it has exited besides logs: I want to run `docker inspect` on it so I can tell why it was killed. All nomad is telling me is `Exit Code: 137, Exit Message: "Docker container exited with non-zero exit code: 137"` which isn't OOMKilled, (that shows up in the exit message). The logs don't have any hints, so I'm trying to debug further but all the containers disappear as soon as they're mysteriously killed.